pkg/cmd/pipeline/start.go

```diff
@@ -567,3 +581,29 @@ func (opt *startOptions) createPipelineResource(resName string, resType v1alpha1
 	fmt.Fprintf(opt.stream.Out, "New %s resource \"%s\" has been created\n", newRes.Spec.Type, newRes.Name)
 	return newRes, nil
 }
+
+func pipelineRunDryRun(output string, s *cli.Stream, pr *v1alpha1.PipelineRun) error {
+	format := strings.ToLower(output)
+
+	if format != "" && format != "json" && format != "yaml" {
+		return fmt.Errorf("output format specifed is %s but must be yaml or json", output)
+	}
+
+	if format == "" || format == "yaml" {
+		prBytes, err := yaml.Marshal(pr)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(s.Out, "%s\n", prBytes)
+	}
+
+	if format == "json" {
+		prBytes, err := json.MarshalIndent(pr, "", "\t")
```

Comment:
Can do something besides tab for indent if there are thoughts on that.

Reply:
no idea :sweat_smile: 
I think `kubectl` prints it without `indent` and let the user pass it through a tool for indent ?

/cc @chmouel

Reply:
I guess I would like the indent here if the behavior is to print for the end user to see.

Reply:
they are spaces on kubectl output or at least according to whitespace-mode in emacs

as long it's yamllint compatible i think it's fine  for me!

> I guess I would like the indent here if the behavior is to print for the end user to see.

It's fine either way? printed to end user or not we could do indent?

Reply:
Without the indent, it's all crammed to the left like below:

```json
{
"kind": "PipelineRun",
"apiVersion": "tekton.dev/v1alpha1",
"metadata": {
"generateName": "deploy-pipeline-run-",
"namespace": "default",
"creationTimestamp": null
},
"spec": {
"pipelineRef": {
"name": "deploy-pipeline"
},
"resources": [
{
"name": "app-git",
"resourceRef": {
"name": "nodejs-ex-git"
}
},
{
"name": "app-image",
"resourceRef": {
"name": "nodejs-ex-image"
}
}
],
"serviceAccountName": "default",
"podTemplate": {}
},
"status": {}
}
```

With the indent, it's as follows:

```json
{
	"kind": "PipelineRun",
	"apiVersion": "tekton.dev/v1alpha1",
	"metadata": {
		"generateName": "deploy-pipeline-run-",
		"namespace": "default",
		"creationTimestamp": null
	},
	"spec": {
		"pipelineRef": {
			"name": "deploy-pipeline"
		},
		"resources": [
			{
				"name": "app-git",
				"resourceRef": {
					"name": "nodejs-ex-git"
				}
			},
			{
				"name": "app-image",
				"resourceRef": {
					"name": "nodejs-ex-image"
				}
			}
		],
		"serviceAccountName": "default",
		"podTemplate": {}
	},
	"status": {}
}
```

I think I'll keep as is.

Reply:
yes please big +1 on indents! any chanced to check if yamllint can validate it ? (i mean not a test but manually)

Reply:
Yes, it works with yamllint.